From f8b59e25927c4b06a3feb7117145dc1ddcc7233d Mon Sep 17 00:00:00 2001
From: Devarsh Thakkar <devarsh.thakkar@xilinx.com>
Date: Wed, 7 Aug 2019 11:44:32 -0700
Subject: [PATCH 0654/1851] Revert "dma: xilinx: Release buffers before DMA
 transfer"

commit 09482b03d7568cfdd4f8f861ceb921124eb80476 from
https://github.com/Xilinx/linux-xlnx.git

This reverts commit 6ce7c27b25efa3e26eb79194b74f7b86efe9ac8f.

The dma driver should be agnostic to low latency capture mode as
it is specific to V4L2 layer. This patch reverts the low latency
related implementation in dma driver.

Signed-off-by: Devarsh Thakkar <devarsh.thakkar@xilinx.com>
Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_frmbuf.c | 13 -------------
 include/linux/dma/xilinx_frmbuf.h  |  1 -
 2 files changed, 14 deletions(-)

diff --git a/drivers/dma/xilinx/xilinx_frmbuf.c b/drivers/dma/xilinx/xilinx_frmbuf.c
index a135af17129b..2545063c9b70 100644
--- a/drivers/dma/xilinx/xilinx_frmbuf.c
+++ b/drivers/dma/xilinx/xilinx_frmbuf.c
@@ -1096,19 +1096,6 @@ static void xilinx_frmbuf_start_transfer(struct xilinx_frmbuf_chan *chan)
 				struct xilinx_frmbuf_tx_descriptor,
 				node);
 
-	if (desc->earlycb == EARLY_CALLBACK_LOW_LATENCY) {
-		dma_async_tx_callback callback;
-		void *callback_param;
-
-		callback = desc->async_tx.callback;
-		callback_param = desc->async_tx.callback_param;
-		if (callback) {
-			callback(callback_param);
-			desc->async_tx.callback = NULL;
-			chan->active_desc = desc;
-		}
-	}
-
 	/* Start the transfer */
 	chan->write_addr(chan, XILINX_FRMBUF_ADDR_OFFSET,
 			 desc->hw.luma_plane_addr);
diff --git a/include/linux/dma/xilinx_frmbuf.h b/include/linux/dma/xilinx_frmbuf.h
index 5307113efd5b..739ccccfcac4 100644
--- a/include/linux/dma/xilinx_frmbuf.h
+++ b/include/linux/dma/xilinx_frmbuf.h
@@ -16,7 +16,6 @@
 
 /* Modes to enable early callback */
 #define EARLY_CALLBACK			BIT(1) /* To avoid first frame delay */
-#define EARLY_CALLBACK_LOW_LATENCY	BIT(2) /* Low latency mode */
 
 /**
  * enum vid_frmwork_type - Linux video framework type
-- 
2.31.1

